Fix CI tests issue with

Fix failing test:
```
=========================== short test summary info ============================
FAILED tests/ignite/metrics/test_roc_auc.py::test_check_compute_fn - Failed: DID NOT WARN. No warnings of type (<class 'ignite.metrics.epoch_metric.EpochMetricWarning'>,) were emitted.
 Emitted warnings: [ UndefinedMetricWarning('Only one class is present in y_true. ROC AUC score is not defined in that case.'),
  UndefinedMetricWarning('Only one class is present in y_true. ROC AUC score is not defined in that case.'),
  UndefinedMetricWarning('Only one class is present in y_true. ROC AUC score is not defined in that case.'),
  UndefinedMetricWarning('Only one class is present in y_true. ROC AUC score is not defined in that case.'),
  UndefinedMetricWarning('Only one class is present in y_true. ROC AUC score is not defined in that case.'),
  UndefinedMetricWarning('Only one class is present in y_true. ROC AUC score is not defined in that case.'),
  UndefinedMetricWarning('Only one class is present in y_true. ROC AUC score is not defined in that case.'),
  UndefinedMetricWarning('Only one class is present in y_true. ROC AUC score is not defined in that case.'),
  UndefinedMetricWarning('Only one class is present in y_true. ROC AUC score is not defined in that case.'),
  UndefinedMetricWarning('Only one class is present in y_true. ROC AUC score is not defined in that case.'),
  UndefinedMetricWarning('Only one class is present in y_true. ROC AUC score is not defined in that case.'),
  UndefinedMetricWarning('Only one class is present in y_true. ROC AUC score is not defined in that case.'),
  UndefinedMetricWarning('Only one class is present in y_true. ROC AUC score is not defined in that case.')].
======================== 1 failed, 14 warnings in 3.43s ========================
```

Most probably the reason of the failure is the new version of scikit-learn: 1.16.0 released 16h ago.
Let's check the test and adapt the warning type to the one that is raised by scikit-learn.


Reported : 
- https://github.com/pytorch/ignite/actions/runs/12246805879/job/34163433521

--- a/tests/ignite/metrics/test_roc_auc.py
+++ b/tests/ignite/metrics/test_roc_auc.py
@@ -4,6 +4,7 @@ from unittest.mock import patch
 import pytest
 import sklearn
 import torch
+from sklearn.exceptions import UndefinedMetricWarning
 from sklearn.metrics import roc_auc_score
 
 import ignite.distributed as idist
@@ -112,7 +113,7 @@ def test_check_compute_fn():
     em = ROC_AUC(check_compute_fn=True)
 
     em.reset()
-    with pytest.warns(EpochMetricWarning, match=r"Probably, there can be a problem with `compute_fn`"):
+    with pytest.warns((UndefinedMetricWarning, EpochMetricWarning), match=r"Only one class.+present in y_true"):
         em.update(output)
 
     em = ROC_AUC(check_compute_fn=False)

